infrastructure/cli/pkg/core/hosts.go

```diff
@@ -0,0 +1,47 @@
+package core
+
+import "github.com/spf13/viper"
+
+type AvailableHosts struct {
```

Comment:
I changed this interface from a `map` since other commands, such as the `ui` command need to be able to clearly tell endpoints apart.